helpers/auth_x11.c

```diff
@@ -77,8 +81,86 @@ int prompt_timeout;
 //! Extra line spacing.
 #define LINE_SPACING 4
 
-//! Whether password display should hide the length.
-int paranoid_password;
+//! Deprecated flag for setting Whether password display should hide the length.
+int paranoid_password_flag;
+
+//! Updated flag for password display choice
+const char * password_prompt_flag;
+
+//! Actual password prompt selected
+enum PasswordPrompt {
+  PASSWORD_PROMPT_CURSOR,
+  PASSWORD_PROMPT_ASTERISKS,
+  PASSWORD_PROMPT_HIDDEN,
+  PASSWORD_PROMPT_DISCO,
+  PASSWORD_PROMPT_EMOJI,
+  PASSWORD_PROMPT_EMOTICON,
+  PASSWORD_PROMPT_KAOMOJI,
+
+  PASSWORD_PROMPT_MAX_VALUE = PASSWORD_PROMPT_KAOMOJI,
+};
+const char *PasswordPromptStrings[] = {
+  /* PASSWORD_PROMPT_CURSOR= */ "cursor",
+  /* PASSWORD_PROMPT_ASTERISKS= */ "asterisks",
+  /* PASSWORD_PROMPT_HIDDEN= */ "hidden",
+  /* PASSWORD_PROMPT_DISCO= */ "disco",
+  /* PASSWORD_PROMPT_EMOJI= */ "emoji",
+  /* PASSWORD_PROMPT_EMOTICON= */ "emoticon",
+  /* PASSWORD_PROMPT_KAOMOJI= */ "kaomoji",
+};
+
+enum PasswordPrompt password_prompt;
+
+// A disco password is composed of multiple disco_dancers (each selected at random from the array), joined by the disco_combiner
+const char * disco_combiner = " ♪ ";
+// Note: the disco_dancers MUST all have the same length
+const char * disco_dancers[] = {
+  "┏(･o･)┛",
+  "┗(･o･)┓",
+};
+
+// Emojis to display in emoji mode. The length of the array must be equal to PARANOID_PASSWORD_LENGTH.
+// List taken from the top items in http://emojitracker.com/
+// The first item is always display in an empty prompt (before typing in the password)
+const char * emojis[] = {
+  "_____", "\U0001f602", "\U00002764", "\U0000267b",
```

Comment:
You've put all the other Unicode characters in there directly - why not these? I.e. actually put the emoji in here as UTF-8, like in the others. More readable.

Reply:
Done in fecdb1b.